File: app/src/main/java/com/lazysoul/kotlinwithandroid/injection/module/ApplicationModule.java
package com.lazysoul.kotlinwithandroid.injection.module;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Lazysoul on 2017. 7. 17..
 */

@Module
public class ApplicationModule {

    private Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return application;
    }

    @Provides
    SharedPreferences provideSharedPrefs() {
        return application.getSharedPreferences("kotlinWithAndroid", Context.MODE_PRIVATE);
    }

    @Provides
    SharedPreferences.Editor provideEditor(SharedPreferences pref) {
        return pref.edit();
    }

}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/injection/module/ApplicationModule.kt
package com.lazysoul.kotlinwithandroid.injection.module

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Created by Lazysoul on 2017. 7. 17..
 */

@Module
class ApplicationModule(private val application: Application) {

    @Provides
    @Singleton
    internal fun provideApplication(): Application {
        return application
    }

    @Provides
    internal fun provideSharedPrefs(): SharedPreferences {
        return application.getSharedPreferences("kotlinWithAndroid", Context.MODE_PRIVATE)
    }

    @Provides
    internal fun provideEditor(pref: SharedPreferences): SharedPreferences.Editor {
        return pref.edit()
    }

}
